src/js/features/approval/saga.js

```diff
@@ -10,14 +10,15 @@ import { APPROVE_TOKEN, APPROVE_PRE_SUCCEEDED, APPROVE_SUCCEEDED, APPROVE_FAILED
 import OwnedUpgradeabilityProxy from '../../../embarkArtifacts/contracts/OwnedUpgradeabilityProxy';
 Escrow.options.address = OwnedUpgradeabilityProxy.options.address;
 
-const {toBN} = web3.utils;
+const {toBN, toWei} = web3.utils;
 
 export function *approveToken({amount}) {
   const feeMilliPercent = yield Escrow.methods.feeMilliPercent().call();
+  const amountWei = toBN(toWei(amount));
```

Comment:
shouldnt we use toTokenDecimals in case the token doesn't use 18 decimals?

Reply:
Ah yes, forgot about that

Reply:
Updated